docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,259 @@
+   
+   The files provide these capabilities.
+      
+   - `PeripheralNames.h` describes the available peripherals and their base addresses.
+   - `PeripheralPins.c` describes the available pins and their association with peripherals.
+   - `PinNames.h` sets macros for pins that define their function.
+   - `system_clock.c` vendor specific file that initializes the system and sets up the clocks.
+      
+1. Modify the files.
+
+   `PinNames.h` is the most common file to be edited. For our example, the ImaginaryBoard is using I2C but connected to different supported signals. Change the I2C pin macro definitions from:
+   
+   ```
+   I2C_SCL     = D15,
+   I2C_SDA     = D14,
+   ```
+   
+   to
+   
+   ```
+   I2C_SCL     = PC_0,
+   I2C_SDA     = PC_1,
+   ```
+   
+   This edit is mainly for convenience if your application code uses standard I2C pin names `I2C_SDA` and `I2C_SCL`.
+   
+   You may also choose to add or remove peripherals, add or remove pins or change the clock frequency by editing `PeripheralNames.h`, `PeripheralPins.c` or `system_clock.c`. In our example, for simplicity, we will not edit these files.
+   
+1. (Optional) Add additional files.
+
+   If necessary, add additional source files for drivers or middleware you have implemented for the new board. In our example, we do not have any files to add.
+
+1. (Optional) Add a simple application source file for testing.
+
+    To confirm the software builds for the new target, add a file named `main.cpp` with the following contents:
+
+    ```  
+    #include "mbed.h"
+    
+    DigitalOut led1(LED1);
+
+    int main()
+    {
+        while (true) {
+            led1 = !led1;
+            wait_ms(500);
+        }
+    }
+    ``` 
+
+    This blinks an LED. Just make sure LED1 is defined for the board.
+    
+    Your directory now looks something like this:
+    
+    ```
+    main.cpp
+    custom_target.json
+    TARGET_IMAGINARYBOARD
+    mbed-os
+    .mbed
+    mbed_settings.py
+    mbed-os.lib
+    ```
+    
+1. Compile the project.
+   
+  With the target configuration set and files added, it is ready to compile. Run the command:
+
+   ```
+   mbed compile -m IMAGINARYBOARD -t <toolchain>
+   ```
+   
+   When successful, it compiles, links and generates a `.bin` file (or `.hex` file for some other boards).
+
+   For example, it prints to the screen:
+   
+   ```
+   Image: .\BUILD\IMAGINARYBOARD\GCC_ARM\mbed-os-imaginary-port.bin
+   ```
+
+1. Program the board.  
+
+  You can test this simple example using a `DISCO-L475VG-IOT01A`. If you actually created a `ImaginaryBoard` board, you could use that too.  
+  
+  <span class="notes">**Note:** Unless your board has an Mbed Enabled debug interface, you need a method of flashing the memory on your board. </span>
+  
+  Since the `DISCO-L475VG-IOT01A` has a Mbed Enabled debug interface (STLink in this case), we can use drag-and-drop programming to flash the board.
```

Comment:
```suggestion
  Since the `DISCO-L475VG-IOT01A` has an Mbed Enabled debug interface (STLink in this case), we can use drag-and-drop programming to flash the board.
```

Reply:
Are we flipping a bit between "We" and "You"?